src/codegen/functions/function.py

```diff
@@ -38,7 +39,7 @@ def __init__(self, function_name, function_metadata):
                     self._parameters.append(FunctionParameter(parameter))
 
                     if parameter["direction"] == "out":
-                        self._output_parameters.append(self._parameters)
+                        self._output_parameters.extend(self._parameters)
```

Comment:
This adds all of the parameters to `self._output_parameters`. I think you want just the last one, right?
```
self._output_parameters.append(self._parameters[-1])
```

Reply:
The intention for using extend was because append was creating list of lists, because _parameters is a list and the whole list got added as list element instead of the only one parameter.

Your suggestion works better, will add the lastly added parameter if its 'out'